msal/application.py

```diff
@@ -597,13 +626,18 @@ def acquire_token_by_username_password(
             - an error response would contain "error" and usually "error_description".
         """
         scopes = decorate_scope(scopes, self.client_id)
+        headers = {'client-request-id': self._get_new_correlation_id(),
+                   'x-client-current-telemetry': self._build_current_telemetry_request_header(
+                       self.ACQUIRE_TOKEN_BY_USERNAME_PASSWORD_ID)}
         if not self.authority.is_adfs:
-            user_realm_result = self.authority.user_realm_discovery(username)
+            user_realm_result = self.authority.user_realm_discovery(username, headers)
```

Comment:
Question: Does our API IDs doc define the User Realm Discovery (URD) to use the same API ID as the AcquireTokenByUsernamePassword one? On the wire, they are 2 different requests. Unless the API ID doc specifically requires us to treat them as the same, I would suggest we can skip telemetry for this undefined URD request, otherwise it would cause our Username Password flow telemetry amount to be doubled. Can you share the API IDs doc link to help us confirm that?

> In the face of ambiguity, refuse the temptation to guess. -- Quoted from the [Zen](https://www.python.org/dev/peps/pep-0020/).

Reply:
Link to API ID doc: https://microsoft.sharepoint.com/:w:/t/aad/devex/EZGF46opilBKrgC59mUFttIBebPBY6MMGLO1eGz0uXQh6w?e=TSq76f

The document does not explicitly answer your question. My 2 cents:  API Id is an identifier for a public API request, so all operations that happen as part of the public API call should have the same public API Id . Having said that, we have updated the telemetry design to only include the telemetry headers to the token endpoint. In this case, we would get rid of  'x-client-current-telemetry' for this request, but IMO we should still include the correlation Id header ('client-request-id'), as when you search for correlation Id in server side logs, you would want to see all requests (including user realm and instance discovery)